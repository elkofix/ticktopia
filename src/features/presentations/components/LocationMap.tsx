export function LocationMap({ latitude, longitude }: { latitude: number; longitude: number }) {
  return (
    <div
      className="bg-white p-6 rounded-xl shadow-md mb-12"
      data-testid="location-map-container"
    >
      <h2 className="text-2xl font-bold text-gray-900 mb-4">Ubicación</h2>
      <div
        className="h-96 w-full rounded-lg overflow-hidden"
        data-testid="map-container"
      >
        <iframe
          width="100%"
          height="100%"
          src={`https://maps.google.com/maps?q=${latitude},${longitude}&z=15&output=embed`}
          title="Ubicación del evento"
          data-testid="map-iframe"
        ></iframe>
      </div>
    </div>
  );
}