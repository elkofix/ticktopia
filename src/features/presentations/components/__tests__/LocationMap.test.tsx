import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { LocationMap } from '../LocationMap';

describe('LocationMap', () => {
  const mockCoords = {
    latitude: 40.7128,
    longitude: -74.0060
  };

  it('renders correctly with provided coordinates', () => {
    render(<LocationMap {...mockCoords} />);

    // Check container
    const container = screen.getByTestId('location-map-container');
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('bg-white');
    expect(container).toHaveClass('p-6');
    expect(container).toHaveClass('rounded-xl');
    expect(container).toHaveClass('shadow-md');
    expect(container).toHaveClass('mb-12');

    // Check title
    const title = screen.getByText('Ubicaci贸n');
    expect(title).toBeInTheDocument();
    expect(title).toHaveClass('text-2xl');
    expect(title).toHaveClass('font-bold');
    expect(title).toHaveClass('text-gray-900');
    expect(title).toHaveClass('mb-4');

    // Check map container
    const mapContainer = screen.getByTestId('map-container');
    expect(mapContainer).toBeInTheDocument();
    expect(mapContainer).toHaveClass('h-96');
    expect(mapContainer).toHaveClass('w-full');
    expect(mapContainer).toHaveClass('rounded-lg');
    expect(mapContainer).toHaveClass('overflow-hidden');

    // Check iframe
    const iframe = screen.getByTitle('Ubicaci贸n del evento');
    expect(iframe).toBeInTheDocument();
    expect(iframe).toHaveAttribute('width', '100%');
    expect(iframe).toHaveAttribute('height', '100%');
    expect(iframe).toHaveAttribute(
      'src',
      `https://maps.google.com/maps?q=${mockCoords.latitude},${mockCoords.longitude}&z=15&output=embed`
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<LocationMap {...mockCoords} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles different coordinate values', () => {
    const newCoords = {
      latitude: 34.0522,
      longitude: -118.2437
    };
    render(<LocationMap {...newCoords} />);

    const iframe = screen.getByTitle('Ubicaci贸n del evento');
    expect(iframe).toHaveAttribute(
      'src',
      `https://maps.google.com/maps?q=${newCoords.latitude},${newCoords.longitude}&z=15&output=embed`
    );
  });

  it('renders with invalid coordinates', () => {
    const invalidCoords = {
      latitude: NaN,
      longitude: NaN
    };
    render(<LocationMap {...invalidCoords} />);

    const iframe = screen.getByTitle('Ubicaci贸n del evento');
    expect(iframe).toHaveAttribute(
      'src',
      'https://maps.google.com/maps?q=NaN,NaN&z=15&output=embed'
    );
  });
});