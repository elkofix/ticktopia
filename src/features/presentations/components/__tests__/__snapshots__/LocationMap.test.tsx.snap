// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationMap matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-6 rounded-xl shadow-md mb-12"
    data-testid="location-map-container"
  >
    <h2
      class="text-2xl font-bold text-gray-900 mb-4"
    >
      Ubicación
    </h2>
    <div
      class="h-96 w-full rounded-lg overflow-hidden"
      data-testid="map-container"
    >
      <iframe
        data-testid="map-iframe"
        height="100%"
        src="https://maps.google.com/maps?q=40.7128,-74.006&z=15&output=embed"
        title="Ubicación del evento"
        width="100%"
      />
    </div>
  </div>
</DocumentFragment>
`;
